storm-client/src/jvm/org/apache/storm/topology/BaseConfigurationDeclarer.java

```diff
@@ -83,4 +83,29 @@ public T setCPULoad(Number amount) {
         }
         return (T) this;
     }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public T addResources(Map<String, Double> resources) {
+        if(resources != null) {
+            return addConfiguration(Config.TOPOLOGY_COMPONENT_RESOURCES_MAP, resources);
+        }
+        return (T) this;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public T addResource(String resourceName, Double resourceValue) {
+        Map<String, Double> resourcesMap = (Map<String, Double>) ((Map<String, Object>) this).get(Config.TOPOLOGY_COMPONENT_RESOURCES_MAP);
+
+        if (resourcesMap != null) {
```

Comment:
I think you mean `if (resourcesMap == null) {` or we are going to be getting NPEs all over the place.